File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {

    let ans={}
    for(let elem of strs){
        let temp=(elem.split("").sort()).join("")
        if(!(temp in ans)){
            ans[temp]=[]
        }
        ans[temp].push(elem)

    }
    return Object.values(ans)
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dec={}
        for item in strs:
            item.replace(" ","")
            key=''.join(sorted(item))
            print(key)
            if key in dec:
                dec[key].append(item)
            else:
                dec[key]=[item]
        res=[]
        for j in dec.values():
            res.append(j)
        res.sort(key=len)
        return res
        

        